Gml 1843 async pytg

luzhoutg opened the pull request.

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1148/

qe-tigergraph reviewed (approved):
QE Approved

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1149/

qe-tigergraph reviewed (approved):
QE Approved

luzhoutg commented:
1. renamed the folder "async_ver2" to "pytgasync" and update all path
2. solved the conflicts with master
3. formatted code
4. added test case using "getVertexCount" and "getEdgeCount"
5. passed all unit tests

parkererickson-tg commented on tmp/Cora/._ingest.gsql:
The temp folder here should be removed

luzhoutg replied:
removed.

parkererickson-tg commented on tests/testserver.json:
does this typo actually work?

luzhoutg replied:
fixed the typo. I modified it after the testing.

parkererickson-tg commented on pyTigerGraph/pytgasync/__init__.py:
This shouldn't contain version or license

luzhoutg replied:
removed

parkererickson-tg replied:
I am saying this shouldn't have __version__ or __license__ at all, should be just the from . import statement. Version and license is covered in `pyTigerGraph/__init__.py`

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1150/

qe-tigergraph reviewed (approved):
QE Approved

RobRossmiller-TG commented on pyTigerGraph/pytgasync/schema.py:
```diff
@@ -0,0 +1,472 @@
+
+
+@dataclass
+class Vertex(object):
```
We don't need to redefine classes that have no changes

luzhoutg commented on pyTigerGraph/pytgasync/pyTigerGraphDataset.py:
```diff
@@ -0,0 +1,91 @@
+        print("---- Ingesting data ----", flush=True)
+        self.graphname = dataset.name
+        if getToken:
+            await self.getToken(self.createSecret())
```
good catch. revise to following:
`if getToken:
     secret = await self.createSecret()
      await self.getToken(secret)`

parkererickson-tg reviewed (commented):
Do we need these functions to be async? These should block before running the loading jobs, as token is a required step before ingest.

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1151/

qe-tigergraph reviewed (approved):
QE Approved

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1152/

qe-tigergraph reviewed (approved):
QE Approved

luzhoutg commented:
- cleaned up the code again to remove some unnecessary pieces.
- kicked off tests, all passed.

RobRossmiller-TG commented on pyTigerGraph/pytgasync/pyTigerGraphAuth.py:
```diff
@@ -0,0 +1,201 @@
+
+logger = logging.getLogger(__name__)
+
+# TODO FINISH MAKING THE ASYNCS AND ADD COMMENTS
```
Is this todo handled?

luzhoutg replied:
This looks like old TODO's. Let me remove it.

RobRossmiller-TG commented on pyTigerGraph/pytgasync/pyTigerGraphAuth.py:
```diff
@@ -0,0 +1,201 @@
+        # uses mainVer instead of _versionGreaterThan4_0 since you need a token for verson checking
+        return res, mainVer
+
+    async def newGetToken(self, secret: str = None, setToken: bool = True, lifetime: int = None) -> Union[tuple, str]:
```
Should this be prefaced with new, or is it just getToken. Same question for refresh and delete

luzhoutg replied:
I reviewed all the TODO comments in the `pytgasync` folder. It appears they were copied directly from the synchronous functions and are outdated, as they haven't been implemented in either the async or sync versions. 

Regarding functions prefixed with "new," it seems Dyllan has broken down the `getToken` function into 'newGetToken' with smaller helper functions: `_prep_newToken`, `_newToken`, and `_parse_newToken`, which can be reused in "newDeleteToken", "newRefreshToken". These new functions are used across unit tests. 

Let me try to comment out "getToken" function, and rename "newGetToken" to "getToken", and run another round test. 

I agree that we might have to reorganize and clean up pyTG as a whole. We can discuss it with the team and add it to sprint planning.

RobRossmiller-TG replied:
So the api is consistent across sync and async, they should not have the "new" prefix

luzhoutg replied:
cleaned up the code by removing new prefix and duplicates, update the functions across different functions.

RobRossmiller-TG reviewed (changes requested):
Check for TODO comments. Not sure if they're old or actual TODO's. Some functions are prefaced with "new" -- why?

This might be a good opportunity to stop and look at pyTG as a whole

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1153/

qe-tigergraph reviewed (approved):
QE Approved

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1154/

qe-tigergraph reviewed (approved):
QE Approved

qe-tigergraph reviewed (commented):
Unit Test: SUCCESS, e2e Test: SKIPPED, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/1158/

qe-tigergraph reviewed (approved):
QE Approved
